### PR TITLE
fix: Add first harvest of session flags to RUM and Trace

### PR DESCRIPTION
**Pull request by cwli24**
Mark the first RUM call and Session Trace harvest of any given session so that it becomes queryable in New Relic.
---

### Overview

Flag in query param of respective request indicating it is the first outgoing payload of a given session.

### Related Issue(s)

NR-166515
NR-166516

### Testing

New test cases added

### CODE REVIEW

File: src/common/session/session-entity.js
Comment (metal-messiah) [inline src/common/session/session-entity.js:144]: This is going to break `isNew` since it uses the undefined check in tandem with line 229 in the `reset()` call

File: src/common/session/session-entity.js
Comment (metal-messiah) [inline src/common/session/session-entity.js:31]: Is this necessary since you will be able to tell that a session `isNew` per page?  only report firstHarvest flag if isNew?

Comment (cwli24) [reply src/common/session/session-entity.js:31]: Falls into the same idea that the SR has with the reason why SentFirstChunk was added -- the real first trace payload can come from any tab theoretically.

This isn't an issue with RUM (flag) however, bc that is unconditionally always sent at load of a new tab, basing off `isNew` makes sense there.

File: src/common/session/session-entity.js
Comment (cwli24) [reply src/common/session/session-entity.js:144]: I removed this bc I think it's extraneous precisely because `reset` will erase the `isNew` property for the `session` manager. Whenever `setup` is ran, `isNew` should be undefined anyways _and_ `isNew` should be set anew in tandem with `initialRead` state.

File: src/common/session/session-entity.js
Comment (metal-messiah) [reply src/common/session/session-entity.js:31]: Just want to double check here -- Is there a race condition though were both tabs could read from session state as `traceHarvestStarted: false` before it has updated, theoretically sending 2 payloads with 

`!this.agentRuntime.session.state.traceHarvestStarted`

Comment (cwli24) [reply src/common/session/session-entity.js:31]: Yeah, so the new `SESSION_EVENTS.UPDATE` just needs to be listened on, will add into stn

Comment (cwli24) [reply src/common/session/session-entity.js:31]: Wait, the session-entity listener to storage updates already manages this. That will update the session.state object.

File: src/features/page_view_event/aggregate/index.js
Comment (metal-messiah) [inline src/features/page_view_event/aggregate/index.js:72]: Everything looks good, just one quick philosophical question here,  Do we want to always send the `fsh` (like shown here), or should we just only send it if its truthy?

**Comment by metal-messiah** [comment]
Approved pending re-run of PR check tests before merging

**Review by metal-messiah — APPROVED** [review]

File: src/features/page_view_event/aggregate/index.js
Comment (cwli24) [reply src/features/page_view_event/aggregate/index.js:72]: If it's not sent at all, there's no session tracking running; distinct from sending '0' ig. That might matter if you're trying to find statistics.

Also does downstream need to be aware of this going as `fsh` before merge? Does it matter what it is as long as they know the exact string?

Comment (metal-messiah) [reply src/features/page_view_event/aggregate/index.js:72]: They have not done the work. It will just have no effect (dropped) until they implement the property on their end

**Pull request merged**
